torch/csrc/jit/passes/shape_analysis.cpp

```diff
@@ -1676,11 +1676,11 @@ class ShapePropagator {
       sizes.at(dim) = length;
       node->output()->setType(tp->withSizesStrides(sizes, tp->strides()));
       return true;
-    } else if (node->matches("aten::sum(Tensor self) -> Tensor")) {
+    } else if (node->matches("aten::sum(Tensor self, *, int? dtype) -> Tensor")) {
```

Comment:
Similarly here in `PropagateCompleteShapeOnNode`, it looks like this code explicitly handles each argument. I think I need to add similar code to extract the `dtype`, but don't have a failing test and I'm not sure how to write one that will reach this code.

Reply:
https://github.com/pytorch/pytorch/blob/master/test/test_jit.py#L6726 here is an example of how to write shape propagation tests